flutter/lib/src/navigation/time_to_initial_display_tracker.dart

```diff
@@ -0,0 +1,135 @@
+
+import '../../sentry_flutter.dart';
+import '../frame_callback_handler.dart';
+
+@internal
+class TimeToInitialDisplayTracker {
+  static final TimeToInitialDisplayTracker _instance =
+      TimeToInitialDisplayTracker._();
+
+  factory TimeToInitialDisplayTracker(
+      {FrameCallbackHandler? frameCallbackHandler}) {
+    if (frameCallbackHandler != null) {
+      _instance._frameCallbackHandler = frameCallbackHandler;
+    }
+    return _instance;
+  }
+
+  TimeToInitialDisplayTracker._();
+
+  FrameCallbackHandler _frameCallbackHandler = DefaultFrameCallbackHandler();
+  bool _isManual = false;
+  Completer<DateTime>? _trackingCompleter;
+  DateTime? _endTimestamp;
+
+  /// This endTimestamp is needed in the [TimeToFullDisplayTracker] class
+  @internal
+  DateTime? get endTimestamp => _endTimestamp;
+
+  Future<void> trackRegularRoute(
+      ISentrySpan transaction,
+      DateTime startTimestamp,
+      String routeName,
+      ) async {
+    await _trackTimeToInitialDisplay(
+      transaction: transaction,
+      startTimestamp: startTimestamp,
+      routeName: routeName,
+      // endTimestamp is null by default, determined inside the private method
+      // origin could be set here if needed, or determined inside the private method
+    );
+  }
+
+  Future<void> trackAppStart(
+      ISentrySpan transaction,
+      AppStartInfo appStartInfo,
+      String routeName,
+      ) async {
+    await _trackTimeToInitialDisplay(
+      transaction: transaction,
+      startTimestamp: appStartInfo.start,
+      routeName: routeName,
+      endTimestamp: appStartInfo.end,
+      origin: SentryTraceOrigins.autoUiTimeToDisplay,
+    );
+
+    // Store the end timestamp for potential use by TTFD tracking
+    _endTimestamp = appStartInfo.end;
+  }
+
+  Future<void> _trackTimeToInitialDisplay({
+    required ISentrySpan transaction,
+    required DateTime startTimestamp,
+    required String routeName,
+    DateTime? endTimestamp,
+    String? origin,
+  }) async {
+    // Determine endTimestamp if not provided
+    final _endTimestamp = endTimestamp ?? await determineEndTime();
+    if (_endTimestamp == null) return;
+
+    final ttidSpan = transaction.startChild(
+      SentrySpanOperations.uiTimeToInitialDisplay,
+      description: '$routeName initial display',
+      startTimestamp: startTimestamp,
+    );
+
+    // Set the origin based on provided value or determine based on _isManual
+    ttidSpan.origin = origin ??
+        (_isManual
+            ? SentryTraceOrigins.manualUiTimeToDisplay
+            : SentryTraceOrigins.autoUiTimeToDisplay);
+
+    final duration = Duration(
+        milliseconds: _endTimestamp.difference(startTimestamp).inMilliseconds);
+    final ttidMeasurement = SentryMeasurement.timeToInitialDisplay(duration);
+
+    transaction.setMeasurement(ttidMeasurement.name, ttidMeasurement.value,
+        unit: ttidMeasurement.unit);
+    await ttidSpan.finish(endTimestamp: _endTimestamp);
```

Comment:
why did you remove the `clear()` call?

Reply:
unless you think we should be more defensive here and clear regardless?

Reply:
I'd be for clearing here, so the logic behind ttid stays in the ttid tracker and not spreadto other classes

Reply:
should we maybe clear at the start and at the end? edge case, because it's also possible that users route to another screen before ttid has finished